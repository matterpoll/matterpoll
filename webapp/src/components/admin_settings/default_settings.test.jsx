import React from 'react';
import {shallow} from 'enzyme';

import DefaultSettings from '@/components/admin_settings/default_settings';

describe('components/admin_settings/DefaultSettings', () => {
    const baseProps = {
        id: 'test id',
        value: {
            anonymous: false,
            anonymousCreator: true,
            progress: false,
            publicAddOption: true,
        },
        label: 'test label',
        disabled: false,
        setByEnv: false,
        config: {},
        license: {},
        onChange: jest.fn(),
        registerSaveAction: jest.fn(),
        setSaveNeeded: jest.fn(),
        unRegisterSaveAction: jest.fn(),
    };

    test('should match snapshot', () => {
        const wrapper = shallow(<DefaultSettings {...baseProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with all true options', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.value = {
            anonymous: true,
            anonymousCreator: true,
            progress: true,
            publicAddOption: true,
        };
        const wrapper = shallow(<DefaultSettings {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with all false options', () => {
        const newProps = {
            ...baseProps,
        };
        newProps.value = {
            anonymous: false,
            anonymousCreator: false,
            progress: false,
            publicAddOption: false,
        };
        const wrapper = shallow(<DefaultSettings {...newProps}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
