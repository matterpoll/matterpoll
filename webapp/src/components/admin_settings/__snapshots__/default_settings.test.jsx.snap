// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_settings/DefaultSettings should match snapshot 1`] = `
<div>
  <DefaultSetting
    label="Don't show who voted for what when the poll ends"
    name="anonymous"
    onChange={[Function]}
    title="Anonymous"
    value={false}
  />
  <DefaultSetting
    label="Don't show author of the poll"
    name="anonymousCreator"
    onChange={[Function]}
    title="Anonymous Creator"
    value={true}
  />
  <DefaultSetting
    label="During the poll, show how many votes each answer option got"
    name="progress"
    onChange={[Function]}
    title="Progress"
    value={false}
  />
  <DefaultSetting
    label="Allow all users to add additional options"
    name="publicAddOption"
    onChange={[Function]}
    title="Public Add Option"
    value={true}
  />
</div>
`;

exports[`components/admin_settings/DefaultSettings should match snapshot with all false options 1`] = `
<div>
  <DefaultSetting
    label="Don't show who voted for what when the poll ends"
    name="anonymous"
    onChange={[Function]}
    title="Anonymous"
    value={false}
  />
  <DefaultSetting
    label="Don't show author of the poll"
    name="anonymousCreator"
    onChange={[Function]}
    title="Anonymous Creator"
    value={false}
  />
  <DefaultSetting
    label="During the poll, show how many votes each answer option got"
    name="progress"
    onChange={[Function]}
    title="Progress"
    value={false}
  />
  <DefaultSetting
    label="Allow all users to add additional options"
    name="publicAddOption"
    onChange={[Function]}
    title="Public Add Option"
    value={false}
  />
</div>
`;

exports[`components/admin_settings/DefaultSettings should match snapshot with all true options 1`] = `
<div>
  <DefaultSetting
    label="Don't show who voted for what when the poll ends"
    name="anonymous"
    onChange={[Function]}
    title="Anonymous"
    value={true}
  />
  <DefaultSetting
    label="Don't show author of the poll"
    name="anonymousCreator"
    onChange={[Function]}
    title="Anonymous Creator"
    value={true}
  />
  <DefaultSetting
    label="During the poll, show how many votes each answer option got"
    name="progress"
    onChange={[Function]}
    title="Progress"
    value={true}
  />
  <DefaultSetting
    label="Allow all users to add additional options"
    name="publicAddOption"
    onChange={[Function]}
    title="Public Add Option"
    value={true}
  />
</div>
`;
